File: js/minafingrar.js
var nånstansDär = function(a) {
	var min = a - 2
	var max = a + 2
	
	if (min < 0) {
        return Math.round(min + Math.random() * (Math.abs(min)+max));
    }else {
        return Math.round(min + Math.random() * max);
    }
}

var hurMycketVägerDen = function() {
	return [8,9,10,11,12]
}

var hurMångaVaDeSmåDvärgar = function(){
	return 7
}

var nuÄrDeDenBiggest = function(){
	return Infinity
}

var grävmaskinerOchHjullastare = function(){
	return ['L90','L150','87..','L180','330','972an','994an']
}

var nåJagHarIngentingAttDölja = function(array){
	return array.map(function(x){
		return true;
	});
}

var vadMenarDuMedDe = function(x){
	return ({}).toString.call(x).match(/\s([a-zA-Z]+)/)[1].toLowerCase()
}

var inte = function(){
	return false
}

var theJnx = function(){
	return ['Hallå Hallå, The Jnx',
	'Hallå där kött och blåbär',
	'Hallåjsan, Hallåjsan',
	'Tjaba Tjena Hallå.. The Jnx',
	'Tjena, Tjena, Tjena']
}

var JaheÅttaKiloMinst = function(){
	return Math.round(8 + Math.random() * 16)+'kg';
}

var ojHanBaraSticker = function(url){
	window.location.replace(url);
}

File: lib/mina_fingrar.rb
require "mina_fingrar/version"
require "execjs"
require "launchy"

class MinaFingrar
  def initialize
    file = File.join(File.dirname(File.expand_path(__FILE__)), '../js/minafingrar.js')
    puts file
    js_file = File.read file

    @js = ExecJS.compile(js_file)
  end

  def js
    @js
  end

  def nånstans_där(a)
    @js.call("nånstansDär", a)
  end

  def hur_mycket_väger_den
    @js.call("hurMycketVägerDen")
  end

  def hur_många_va_de_små_dvärgarna
    @js.call("hurMångaVaDeSmåDvärgar")
  end

  def nu_är_de_den_biggest
    @js.call("nuÄrDeDenBiggest")
  end

  def grävmaskiner_och_hjullastare
    @js.call("grävmaskinerOchHjullastare")
  end

  def nå_jag_har_ingenting_att_dölja(array)
    @js.call("nåJagHarIngentingAttDölja", array)
  end

  def vad_menar_du_med_de(x)
    @js.call("vadMenarDuMedDe",x)
  end

  def inte
    @js.call("inte")
  end

  def the_jnx
    @js.call("theJnx")
  end

  def jahe_åtta_kilo_minst
    @js.call("JaheÅttaKiloMinst")
  end

  def oj_han_bara_sticker(url, opts)
    Launchy.open(url, opts)
  end
end
